tools/run_tests/run_tests.py

```diff
@@ -155,6 +156,8 @@ def configure(self, config, args):
                                                                        self.args.compiler)
 
   def test_specs(self):
+    if self.args.compiler is _FREE_PASS and self.args.travis:
```

Comment:
We are currently seeing this with just Python3.5, 3.6, and PyPy. But I don't see how other languages won't be affected by the same problem in future. The original fix was Python focussed but the conclusion was that the change is better made to all languages for consistent behaviour.